src/components/FilterBar/FilterDisplay/FilterInputFactory.react.js

```diff
@@ -0,0 +1,38 @@
+import {TextInput} from "./Inputs/TextInput.react";
+import {DateInput} from "./Inputs/DateInput.react";
+import {SelectInput} from "./Inputs/SelectInput.react";
+
+export function FilterInputFactory(type, value, uid, key) {
+  // Janky way to ensure uniqueness of the input, so that it re-renders the
+  // value in the input rather than just diffing based on input type.
+
+  key = Date.now();
+
+  if (type === "text" || type === "id") {
```

Comment:
can this be a hash and look up?

klass = type.charAt(0).toUpperCase() + type.slice(1) + "Input";

<{klass} />